### PR TITLE
Add horizontal pod autoscaling v1 and v2beta2 support to gateway-proxy

### PR DESCRIPTION
**Pull request by kevin-shelaga**
# Description

Add horizontal pod autoscaling v1 and v2beta2 support to gateway-proxy

Partially addresses https://github.com/solo-io/gloo/issues/2229

# Context
Add horizontal pod autoscaling v1 and v2beta2 support to gateway-proxy

Differs from PR: https://github.com/solo-io/gloo/pull/3685 in that this PR extends HPA support beyond a basic implementation with a way for other components to leverage.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2229

### CODE REVIEW

**Comment by kevin-shelaga** [comment]
test flake?

File: install/helm/gloo/generate/values.go [hunk 1]
```diff
@@ -319,6 +320,15 @@ type GatewayProxyDeployment struct {
 	*DeploymentSpecSansResources
 }
 
+type HorizontalPodAutoscaler struct {
+	ApiVersion                     string                   `json:"apiVersion,omitempty" desc:"accepts autoscaling/v1 or autoscaling/v2beta2."`
+	MinReplicas                    int32                    `json:"minReplicas,omitempty" desc:"minReplicas is the lower limit for the number of replicas to which the autoscaler can scale down."`
+	MaxReplicas                    int32                    `json:"maxReplicas,omitempty" desc:"maxReplicas is the upper limit for the number of replicas to which the autoscaler can scale up. It cannot be less that minReplicas."`
+	TargetCPUUtilizationPercentage int32                    `json:"targetCPUUtilizationPercentage,omitempty" desc:"target average CPU utilization (represented as a percentage of requested CPU) over all the pods. Used only with apiVersion autoscaling/v1"`
+	Metrics                        []map[string]interface{} `json:"metrics,omitempty" desc:"metrics contains the specifications for which to use to calculate the desired replica count (the maximum replica count across all metrics will be used). Used only with apiVersion autoscaling/v2beta2"`
```
Comment (kdorosh) [inline install/helm/gloo/generate/values.go:328]: did you mean to have `[]map[string]interface{}`? I'm thinking not

**Review by kdorosh — COMMENTED** [review]
one docs nit, otherwise lgtm

File: install/helm/gloo/generate/values.go (hunk 1 above)
Comment (kevin-shelaga) [reply install/helm/gloo/generate/values.go:328]: @kdorosh Metrics is actually a []MetricSpec, so i think []map[string]interface{} might still be appropriate still. Please let me know if you disagree.

https://pkg.go.dev/k8s.io/api/autoscaling/v2beta2#HorizontalPodAutoscalerSpec

```go
        // metrics contains the specifications for which to use to calculate the
	// desired replica count (the maximum replica count across all metrics will
	// be used).  The desired replica count is calculated multiplying the
	// ratio between the target value and the current value by the current
	// number of pods.  Ergo, metrics used must decrease as the pod count is
	// increased, and vice-versa.  See the individual metric source types for
	// more information about how each type of metric must respond.
	// If not set, the default metric will be set to 80% average CPU utilization.
	// +optional
	Metrics []MetricSpec `json:"metrics,omitempty" protobuf:"bytes,4,rep,name=metrics"`
```

**Review by kdorosh — APPROVED** [review]

**Pull request merged**